sub-packages/bionemo-llm/src/bionemo/llm/infer.py

```diff
@@ -0,0 +1,164 @@
+        accumulate_grad_batches=parallel_config.accumulate_grad_batches,
+        tensor_model_parallel_size=parallel_config.tensor_model_parallel_size,
+        pipeline_model_parallel_size=parallel_config.pipeline_model_parallel_size,
+    )
+    # setup megatron strat and setup trainer
+    # verified that the extra args in the trainer we made explicit are irrelevant for inference.
+    strategy = nl.MegatronStrategy(
+        tensor_model_parallel_size=parallel_config.tensor_model_parallel_size,
+        pipeline_model_parallel_size=parallel_config.pipeline_model_parallel_size,
+        ddp="megatron",
+        find_unused_parameters=True,
+        ckpt_include_optimizer=True,
+        progress_interval=1,
+    )
+    trainer = nl.Trainer(
+        devices=parallel_config.num_devices,
+        accelerator="gpu",
+        strategy=strategy,
+        num_nodes=parallel_config.num_nodes,
+        callbacks=[],
+        plugins=nl.MegatronMixedPrecision(precision=precision),
+    )
+
+    # setup datamodule for inference
+    # TODO this doesnt set it up for inference.
+    datamodule: DataModuleT = data_config.construct_data_module(global_batch_size)
+
+    if isinstance(model_config_cls, config_models.ExposedModelConfig):
+        # Get associated type
+        config_class = model_config_cls.model_class()
+    elif isinstance(model_config_cls, model.BioBertConfig):
+        config_class = model_config_cls
+    else:
+       raise ValueError("Expected either ExposedModelConfig or BioBertConfig for bionemo-model-config-cls")
+
+    # atleast BioBertConfig
+    bionemo_model_config = config_class(
+        seq_length=data_config.seq_length,
+        params_dtype=get_autocast_dtype(precision),
+        pipeline_dtype=get_autocast_dtype(precision),
+        autocast_dtype=get_autocast_dtype(precision),  # setting this speeds things up a lot
+        # handle checkpoint resumption here rather than auto-resume so this supports fine-tuning capabilities
+        initial_ckpt_path=str(checkpoint_path) if checkpoint_path is not None else None,
+        include_embeddings=include_embeddings,
+        include_hiddens=include_hiddens,
+        include_input_ids=include_input_ids,
+        skip_logits=not include_logits,
+        initial_ckpt_skip_keys_with_these_prefixes=[],
+    ) # type: ignore
+
+    # setup lightning mod
+    module: BionemoLightningModule = biobert_lightning_module(
+        config=bionemo_model_config, tokenizer=datamodule.tokenizer, optimizer=None
+    )
+    # setup results
+    pathlib.Path(data_config.result_dir).mkdir(parents=True, exist_ok=True)
+    # setup lightning module
+
+    # TODO will want to bring in Farhad's work, eventually we should rebase on his PR
+    # Does this have to be a DataModule or can it be a DataLoader?
+    predictions = trainer.predict(module, datamodule=datamodule, return_predictions=True)
+    # TODO evaluate this section. I worry about gathering predictions in memory.
```

Comment:
So in the future we should be able to say `return_predictions=False` and use a prediction writer. I think you were getting at that with mention of Farhad's PR but maybe specifically say that we should get return_predictions=False working alongside a prediction writer.